subsys/bluetooth/host/l2cap.c

```diff
@@ -331,6 +331,23 @@ void bt_l2cap_chan_add(struct bt_conn *conn, struct bt_l2cap_chan *chan,
 	LOG_DBG("conn %p chan %p", conn, chan);
 }
 
+static void init_le_chan_private(struct bt_l2cap_le_chan *le_chan)
+{
+	/* Initialize private members of the struct. We can't "just memset" as
+	 * some members are used as application parameters.
+	 */
```

Comment:
Outside the scope of this PR, but we could move all stack-owned data into the start or end of the struct (or even put it into a struct within the struct itself). 

That way we can group app vs stack owned memory, and apply `memset` with the stack-owned part or use `offsetof` to memset a subset of the struct